azkaban-common/src/main/java/azkaban/executor/ExecutionController.java

```diff
@@ -249,28 +249,74 @@ public ExecutableFlow getExecutableFlow(final int execId)
   }
 
   /**
-   * Get all active (running, non-dispatched) flows from database. {@inheritDoc}
+   * Get all running (unfinished) flows from database. {@inheritDoc}
    */
   @Override
   public List<ExecutableFlow> getRunningFlows() {
     final ArrayList<ExecutableFlow> flows = new ArrayList<>();
     try {
-      getActiveFlowHelper(flows, this.executorLoader.fetchUnfinishedFlows().values());
+      getFlowsHelper(flows, this.executorLoader.fetchUnfinishedFlows().values());
     } catch (final ExecutorManagerException e) {
       this.logger.error("Failed to get running flows.", e);
     }
     return flows;
   }
 
   /**
-   * Helper method to get all running flows from a Pair<ExecutionReference,
-   * ExecutableFlow collection
+   * Helper method to get all flows from collection.
    */
-  private void getActiveFlowHelper(final ArrayList<ExecutableFlow> flows,
+  private void getFlowsHelper(final ArrayList<ExecutableFlow> flows,
       final Collection<Pair<ExecutionReference, ExecutableFlow>> collection) {
-    for (final Pair<ExecutionReference, ExecutableFlow> ref : collection) {
-      flows.add(ref.getSecond());
+    collection.stream().forEach(ref -> flows.add(ref.getSecond()));
+  }
+
+  /**
+   * Get execution ids of all running (unfinished) flows from database. {@inheritDoc}
+   */
+  @Override
+  public String getRunningFlowIds() {
```

Comment:
why returning a string instead of a list?

Reply:
+1, we can do list->string serialization on the caller side. the function should provide behavior consistent with its name.

Reply:
Changed from string to list, thanks.